code-challenges.test.js

```diff
@@ -35,35 +49,115 @@ const num2 = 0
+ // Give it parameters that include methods (charAt to select the first character and toUpperCase to capitalize the first letter)
+ // .slice will then return the remaining parts of the word. It'll do this using the index of the characters so (1) will return the word from it's second character onwards. 
+
+
+describe("capitalizeMe", () => {
+    it("a function that takes in an array of words and returns an array with all the words capitalized", () => {
+      expect(capitalizeMe(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(capitalizeMe(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+    })
+})  
+  
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
+// ReferenceError: capitalizeMe is not defined
 
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
+const capitalizeMe = (array) => {
+   return array.map(noun => {
+    return noun.charAt(0).toUpperCase() + noun.slice(1)
+  })
+}
 
-// b) Create the function that makes the test pass.
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.462 s, estimated 1 s
 
 
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
-
-
 // b) Create the function that makes the test pass.
+
+// Pseudocode:
+ // Declare a function called firstVowelIndexer
+ // Give it parameters (string)
+ // Create an array inside the function (let vowels) and include the 5 vowels.
+ // Write a for loop that will iterate over the array 
+ // create an if statement using .includes that will return the index of the first vowel in each word
+
+describe("firstVowelIndexer", () => {
+     it("a function that takes in a string and logs the index of the first vowel", () => {
+       expect(firstVowelIndexer(vowelTester1)).toEqual(1)
+       expect(firstVowelIndexer(vowelTester2)).toEqual(0)
+       expect(firstVowelIndexer(vowelTester3)).toEqual(2)
+     })
+   })
+
+ const vowelTester1 = "learn"
+ // Expected output: 1
+ const vowelTester2 = "academy"
+// // Expected output: 0
+ const vowelTester3 = "challenges"
+// // Expected output: 2
+
+
+const firstVowelIndexer = (string) => {
+  let vowels = ["a", "e", "i", "o", "u"]
+  for (let i = 0; i < string.length; i++) {
+    if (vowels.includes(string[i])) {
+      return i
+    }
+  }
+}
```

Comment:
Excellent work!  I also really like that you attempted this with a higher order function.  A suggestion for your tests would be to place your test variables into the it method within your test.  Overall, this section was outstanding!